### PR TITLE
Always pass FORCE_COLOR & NO_COLOR to the environment

### PR DESCRIPTION
**Pull request by hashar**
The environment variables `FORCE_COLOR` and `NO_COLOR`are a popular way to force or disable color output. An example usage is pytest being run under a CI system.

- [x] ran the linter to address style issues (`tox -e fix`)
- [x] wrote descriptive pull request text
- [x] ensured there are test(s) validating the fix
- [x] added news fragment in `docs/changelog` folder
- [ ] updated/extended the documentation

### CODE REVIEW

**Review by gaborbernat — CHANGES_REQUESTED** [review]
https://no-color.org/ and FORCE_COLOR are the standard env vars. These app-specific ones don't get to be in the core.

**Comment by hashar** [comment]
I went with `PY_COLORS` a while ago which was even used by tox at some point ( #163 ).  The other use case was for `pytest` but it has learned about `NO_COLOR` `FORCE_COLOR` since version 6.0.0 ( https://github.com/pytest-dev/pytest/pull/7466 ).
```python
def should_do_markup(file: TextIO) -> bool:
    if os.environ.get("PY_COLORS") == "1":
        return True
    if os.environ.get("PY_COLORS") == "0":
        return False
    if "NO_COLOR" in os.environ:
        return False
    if "FORCE_COLOR" in os.environ:
        return True
    return (
        hasattr(file, "isatty") and file.isatty() and os.environ.get("TERM") != "dumb"
    )
```

May I repurpose this pull request to add `NO_COLOR` and `FORCE_COLOR` to the environment variables passed by default? I find them handy.

**Comment by gaborbernat** [comment]
I don't think you need to add support for them. They should work today. Did you find the opposite?

**Comment by hashar** [comment]
`NO_COLOR` and `FORCE_COLOR` are only handled by the  config cli parser to set the `color` flag. For the environments, they are not passed:
```ini
[testenv]
commands = python3 -c 'import os; print(os.environ.get("FORCE_COLOR"))'
```
```
$ FORCE_COLOR=1 tox -q
None
```

(same for `NO_COLOR`)

**Comment by gaborbernat** [comment]
If that doesn't work, a pull request is welcome. Just make sure to add the test and the changelog.

**Comment by hashar** [comment]
May you please reopen the pull request or should I send a new one? :)

File: src/tox/tox_env/api.py [hunk 1]
```diff
@@ -219,6 +219,7 @@ def _default_pass_env(self) -> list[str]:
             "LD_LIBRARY_PATH",  # location of libs
             "LDFLAGS",  # linker flags
             "HOME",  # needed for `os.path.expanduser()` on non-Windows systems
+            "PY_COLORS",  # Popular way to force color output in CI
```
Comment (gaborbernat) [inline src/tox/tox_env/api.py:222]: Use FORCE_COLOR, NO_COLOR instead.

**Review by gaborbernat — CHANGES_REQUESTED** [review]
Don't forget to add test and changelog.

File: src/tox/tox_env/api.py (hunk 1 above)
Comment (hashar) [reply src/tox/tox_env/api.py]: Done! Replaced by `FORCE_COLOR` and `NO_COLOR`.

**Comment by hashar** [comment]
@gaborbernat I have repurposed the pull request to pass pass `FORCE_COLOR` and `NO_COLOR` instead of `PY_COLORS`.

**Comment by gaborbernat** [comment]
Yeah, but the CI is failing now that's blocking progress.

**Comment by hashar** [comment]
A test was failing under Windows OS. 

I also fixed `docs/changelog/3171.feature.rst` which I forgot to update.

**Review by gaborbernat — APPROVED** [review]

**Pull request merged**